rhino/src/main/java/org/mozilla/javascript/Context.java

```diff
@@ -2513,8 +2501,7 @@ private ScriptNode parse(
             int lineno,
             CompilerEnvirons compilerEnv,
             ErrorReporter compilationErrorReporter,
-            boolean returnFunction)
-            throws IOException {
+            boolean returnFunction) {
```

Comment:
I assume, that in former time a Reader was passed instead of `String sourceString` and that's why the IOException was still there. It can be removed and in all depending calls also.

Note: This might be a breaking change, if someone has implemented his own context with overwritten `compileImpl`